### PR TITLE
Update format settings

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomvanmele** [comment]
check the first commit for the settings

**Comment by tomvanmele** [comment]
```toml
[tool.black]
line-length = 179

[tool.ruff]
line-length = 179
indent-width = 4
target-version = "py39"

[tool.ruff.lint]
select = ["E", "F", "I"]

[tool.ruff.lint.pycodestyle]
max-doc-length = 179
```

File: pyproject.toml [hunk 1]
```diff
@@ -118,20 +118,15 @@ replace = "[{new_version}] {now:%Y-%m-%d}"
 # ============================================================================
 
 [tool.black]
-# we should set this to 179
-line-length = 120
+line-length = 179
```
Comment (gonzalocasas) [inline pyproject.toml:121]: There's a paragraph in the devguide that needs to be updated to reflect this change as well. And we need to create follow up issues to update this across the ecosystem (compas theme, and other core packages)

File: pyproject.toml [hunk 2]
```diff
@@ -118,20 +118,15 @@ replace = "[{new_version}] {now:%Y-%m-%d}"
...
 indent-width = 4
 target-version = "py39"
 
 [tool.ruff.lint]
-# we should add "I"
-select = ["E", "F"]
-# we should remove this
-ignore = ["E501"]
+select = ["E", "F", "I"]
```
Comment (gonzalocasas) [inline pyproject.toml:129]: Without having to read the ruff docs, does this imply we enable all? What is excluded? Maybe drop a line indicating why/what is done here

**Review by gonzalocasas — APPROVED** [review]
Besides a tiny comment, it lgtm!

I do wonder though, why raising the max line length? Any concrete reason?

File: pyproject.toml (hunk 1 above)
Comment (gonzalocasas) [reply pyproject.toml:121]: In that line, I think the devguide needs to be update to indicate the move to `ruff`, right?

File: pyproject.toml (hunk 2 above)
Comment (tomvanmele) [reply pyproject.toml:129]: there is also "B", "Q", ... which are things we don't need/want

File: pyproject.toml (hunk 1 above)
Comment (tomvanmele) [reply pyproject.toml:121]: > There's a paragraph in the devguide that needs to be updated to reflect this change as well. And we need to create follow up issues to update this across the ecosystem (compas theme, and other core packages)

am already doing this for `compas_cgal`, `compas_occ`, `compas_libigl`, `compas_gmsh`, `compas_viewer`, `compas_notebook`, `compas_model`

Comment (tomvanmele) [reply pyproject.toml:121]: re the devguide: yes indeed, we should. will make the change...

**Comment by tomvanmele** [comment]
> Besides a tiny comment, it lgtm!
> 
> I do wonder though, why raising the max line length? Any concrete reason?

i just very often bump into the fact that 120 is too short. many comprehensions, or nested function calls get formatted really awkwardly because of it. also in docstrings it happens a lot (ruff doesn't properly differentiate those afaik). if you don't like it we can change it back :)

**Comment by gonzalocasas** [comment]
> > Besides a tiny comment, it lgtm!
> > I do wonder though, why raising the max line length? Any concrete reason?
> 
> i just very often bump into the fact that 120 is too short. many comprehensions, or nested function calls get formatted really awkwardly because of it. also in docstrings it happens a lot (ruff doesn't properly differentiate those afaik). if you don't like it we can change it back :)

nah, it's fine, I wasn't sure if there was an underlying technical reason for it (like ruff defaulting to it, or something like that)

**Pull request merged**